app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt

```diff
@@ -0,0 +1,42 @@
+package org.oppia.app.home
+
+import android.content.Intent
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.drawerlayout.widget.DrawerLayout
+import org.oppia.app.R
+import org.oppia.app.activity.ActivityScope
+import org.oppia.app.drawer.NavigationDrawerFragment
+import javax.inject.Inject
+
+/** The controller for [HomeActivityPresenter]. */
+@ActivityScope
+class HomeActivityPresenter @Inject constructor(private val activity: AppCompatActivity) {
+  var toolbar: Toolbar? = null
+  var navigationDrawerFragment: NavigationDrawerFragment? = null
+
+  fun init(title: String) {
+    toolbar = activity.findViewById<View>(R.id.toolbar) as Toolbar?
```

Comment:
For this & ``navigationDrawerFragment``--do we actually need to save these as fields? If not, prefer not to. Every time I see a saved Android layout/view/fragment in a field, I have to do a mental check to make sure it's not being unintentionally leaked. In general, we should try to avoid storing these in fields unless absolutely necessary to make our UIs more robust and less likely to fail in difficult-to-debug ways.

Reply:
Is it okay to save it as private

Reply:
``private`` is definitely better than the alternative, but why do these need to be fields at all? They don't seem to be reused outside of ``init``.

Reply:
updated it as local val inside init.